### PR TITLE
added listener for Map/Survey which reloads page state everytime tab …

### PR DESCRIPTION
**Pull request by fchau-arrdude**
added code to reload state if view is accessed from the bottom bar

### CODE REVIEW

File: src/screens/Map.js [hunk 1]
```diff
@@ -15,22 +15,42 @@ export default class Map extends Component {
     
         }
       }
+
+      resetState() {
```
Comment (vvvli) [inline src/screens/Map.js]: What is the plan with this function?

File: src/screens/SliderInput.js [hunk 2]
```diff
@@ -15,14 +15,30 @@ export default class ClassSlider extends Component {
...
+    console.log(ind)
+    this.setState({index: ind});
+    console.log(this.state)
+  }
+
   async componentDidMount() {
-    let survey = await API.get('surveysCRUD', `/surveys/random`);
-    this.setState({ survey, loading: false });
+    this._sub = this.props.navigation.addListener(
+      'didFocus',
```
Comment (vvvli) [inline src/screens/SliderInput.js:33]: What does 'didFocus' mean?

**Review by vvvli — COMMENTED** [review]
I see the wellness listener implementation and added some comments wondering what a few lines mean. Another question is the forceRemount function necessary?

File: src/screens/SliderInput.js (hunk 2 above)
Comment (fallguy) [reply src/screens/SliderInput.js:33]: awesome comments @vvvli !

File: src/screens/Map.js (hunk 1 above)
Comment (fchau-arrdude) [reply src/screens/Map.js]: Not used. will remove

File: src/screens/SliderInput.js (hunk 2 above)
Comment (fchau-arrdude) [reply src/screens/SliderInput.js:33]: DidFocus is a listener that detects if the 'tab' is 'focused'. Basically, when a user taps on an icon, that is a focus.

**Pull request merged**